src/lib.rs

```diff
@@ -426,41 +547,42 @@ impl Default for SymmetricKey {
     }
 }
 
-impl fmt::Display for PublicKeys {
+#[derive(Serialize, Deserialize)]
+struct CipherText {
+    nonce: [u8; box_::NONCEBYTES],
+    ciphertext: Vec<u8>,
+}
+
+impl fmt::Display for PublicEncryptKey {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        for b in &self.sign[..] {
-            write!(f, "{:02x}", b)?;
-        }
-        for b in &self.encrypt[..] {
-            write!(f, "{:02x}", b)?;
-        }
+        write!(
+            f,
+            "{:02x}{:02x}{:02x}..",
+            &self.encrypt.0[0], &self.encrypt.0[1], &self.encrypt.0[2]
+        )?;
```

Comment:
If you remove `?;` here, you don't need to return `Ok(())` below.  Same for other two `impl fmt::Display`s below.